lib/net.js

```diff
@@ -766,6 +773,15 @@ function connect(self, address, port, addressType, localAddress, localPort) {
 
   var err;
 
+  if (self._setNoDelay)
```

Comment:
I don't think you need this `if`. And if `self._setNoDelay` is `false`, this will never get executed.